### ISSUE TITLE
Game crashes when module is enabled

### ISSUE THREAD

**Issue by getsmp** [Terasology/Fences#3]
Game version: 1311
Module version: 0.1.0
Log with only Fences enabled: http://pastebin.com/v457ULKf

**Comment by Cervator** [comment]
Peculiar, still no obvious issue in the log, other than yeah, a duplicate Core entry, which is weird.

Can you try the following:

1. Download latest stable: http://jenkins.terasology.org/job/DistroOmegaRelease/lastSuccessfulBuild/artifact/distros/omega/build/distributions/TerasologyOmega.zip
1. Put it in a new location you haven't used before, like `/home/nick/Terasology`
1. Launch the game via command line using `java -jar Terasology.jar -homedir`
1. See if you get the duplicate core warning in the log, see if the game runs without Fences, see if the game runs with Fences

Thanks for all your effort troubleshooting this by the way :-)

**Comment by getsmp** [comment]
You didn't understand me. Occasionally I copied two Core modules of different versions. I removed the oldest of them. So forget about duplicate Core module ;)

Next, I have tested game with every module enabling them one by one. The game worked fine without Fences enabled (and all other modules enabled), but enabling Fences triggered a crash. The log I attached was the clean log from clean installation, because I always clear log folder after every run for better result.

So, I have only to test if the game crashes when using stable branch. But.. I think there still won't be any errors in the log. Is there any module-testing tool or debug mode?

**Comment by getsmp** [comment]
I followed your instructions and get Terasology (stable) working with Fences enabled. THank you! But then issue is in testing branch...

**Comment by Cervator** [comment]
We might be able to trigger some more debug logging, yeah. But running with all modules is *highly* discouraged as they aren't all compatible with each other. Generally it is safe to use the gameplay templates as those are actually tested, and adding a few more independent things (like WildAnimals) on top is usually fine. But not everything :-)

**Comment by Cervator** [comment]
Re-opening this as it still seems to be possible to replicate - somehow :(

Linking to https://github.com/MovingBlocks/Terasology/issues/1539 as it is really the only idea I've got. Maybe we can retest after fixing some math.

@crazywolf132 can replicate with latest dev build (engine build 1324, omega 54) with just Fences present

**Comment by Cervator** [comment]
Actually, I think I identified it - some of the modules, including Fences, hadn't been rebuilt since the TeraMath merge to engine. After rebuilding just that module @crazywolf132 was able to run the game with Fences active and no crash.

I've rebuilt all our modules and redone an Omega zip (develop version, not stable). @getsmp are you still around and able to test http://jenkins.terasology.org/job/DistroOmega/lastSuccessfulBuild/artifact/distros/omega/build/distributions/TerasologyOmega.zip ?

I'm going to close this again as presumed fixed
